src/App.tsx

```diff
@@ -1,61 +1,148 @@
 import './App.scss';
+import { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/todo';
+import { User } from './types/user';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const getNewUser = (userId: number): User | null => {
+  return usersFromServer.find(
+    user => user.id === userId,
+  ) || null;
+};
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getNewUser(todo.userId),
+}));
 
 export const App = () => {
+  const [newTitle, setNewTitle] = useState('');
+  const [newUserId, setNewUserId] = useState(0);
+  const [errorTitleMessage, setErrorTitleMessage] = useState(false);
+  const [errorUserMessage, setErrorUserMessage] = useState(false);
+  const [newTodos, setNewTodos] = useState(todos);
+
+  const addNewTodo = () => {
+    if (newUserId && newTitle) {
+      const newTodo = {
+        id: Math.max(...newTodos.map(todo => todo.id)) + 1,
+        userId: newUserId,
+        completed: false,
+        title: newTitle,
+        user: getNewUser(newUserId),
+      };
+
+      setNewTodos((prevTodos) => ([...prevTodos, newTodo]));
+      setNewUserId(0);
+      setNewTitle('');
+      setErrorTitleMessage(false);
+      setErrorUserMessage(false);
+    }
+
+    if (!newTitle) {
+      setErrorTitleMessage(true);
+    }
+
+    if (!newUserId) {
+      setErrorUserMessage(true);
+    }
+  };
+
+  const isSubmitted = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (newUserId && newTitle && newTitle.trim()) {
+      addNewTodo();
+    } else {
+      if (!newTitle || !newTitle.trim()) {
+        setErrorTitleMessage(true);
+      } else {
+        setErrorTitleMessage(false);
+      }
+
+      if (!newUserId) {
+        setErrorUserMessage(true);
+      } else {
+        setErrorUserMessage(false);
+      }
```

Comment:
try to avoid nested if/else statements
it can be something like that
```
if (!newTitle.trim()) {
   setErrorTitleMessage(true);
}

if (!newUserId) {
  setErrorUserMessage(true);
}

if ( !newTitle.trim() || !newUserId) {
  return
}

addNewTodo();
```
set error states on false better in onFocus event or onCnange event